mempool/clist_mempool.go

```diff
@@ -643,9 +644,37 @@ type mempoolTx struct {
 	gasWanted int64    // amount of gas this tx states it will require
 	tx        types.Tx //
 
-	// ids of peers who've sent us this tx (as a map for quick lookups).
-	// senders: PeerID -> bool
-	senders sync.Map
+	// mu guards following fields
+	mu sync.Mutex
+	// senders is a bitset ids of peers who've sent us this tx
+	// TODO: ready libs/bits.BitArray for
+	// zero value use, then use it instead.
+	senders big.Int
```

Comment:
if senderID is close to max uint16 it will allocate 8kb per mempoolTx, which will definitely make the benchmark slower.
not a risk if senderID is a counter, but could be if it is not

Reply:
Ah, fair point. So maybe a plain `map[uint16]struct{}` is ok, let me try.

Reply:
@dshulyak PTAL!